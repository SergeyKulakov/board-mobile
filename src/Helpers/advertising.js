export const getDataWithAds = data => {
  const result = []

  data.forEach((el, key) => {
    const index = key + 1
    if (data[key + 1]) {
      if (index % 5 === 0) {
        result.push(el)
        result.push('ads')
      } else if (index === 3 || index % 7 === 0) {
        result.push(el)
        result.push('sponsor')
      } else result.push(el)
    } else result.push(el)
  })

  return result
}
