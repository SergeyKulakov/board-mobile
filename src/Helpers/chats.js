import moment from 'moment'

export const getMessageTime = date => {
  const TODAY = moment()
  const YESTERDAY = moment().date(TODAY.get('date') - 1)

  const data = Number(date)

  if (moment(data).isSame(TODAY, 'day')) return moment(data).format('HH:mm')
  if (moment(data).isSame(YESTERDAY, 'day')) return 'Yesterday'
  return moment(data).format('DD MMM YYYY')
}
