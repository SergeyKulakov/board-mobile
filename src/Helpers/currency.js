import data from 'Constants/country-currency-codes'

export const getSearchData = string => {
  if (!string || typeof string !== 'string') return data

  const value = string.toLowerCase()

  return data.filter(
    el =>
      el.country.toLowerCase().indexOf(value) !== -1 ||
      el.country.toLowerCase().indexOf(value) !== -1 ||
      el.code.toLowerCase().indexOf(value) !== -1,
  )
}

export default data
