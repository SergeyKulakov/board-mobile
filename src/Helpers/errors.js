import _ from 'lodash'
import i18n from 'I18N'

export const getError = (error, path = 'apiErrors') => {
  const message = _.get(error, 'payload.message')
  const code = _.get(error, 'payload.code')

  if (
    error.payload === 'Internal server error' ||
    message === 'Internal server error' ||
    message === 'Internal Server Error' ||
    code === 'internal'
  )
    return i18n.t('apiErrors.InternalServerError')

  if (error.message === 'Unauthorized' || error.payload === 'Unauthorized') {
    return i18n.t('apiErrors.unauthorizedError')
  }

  if (!code) return `${message || error.payload}`

  return i18n.t(`${path}.${code}`)
}
