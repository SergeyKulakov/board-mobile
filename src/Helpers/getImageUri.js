import { BACKET } from 'react-native-dotenv'
import ProfileImagesManager from 'Services/Images/ProfileImagesManager'
import { profileImageTypes } from 'Services/Constants/profileImages.constants'
import _ from 'lodash'
import isUrl from 'is-url'

const s3 = 'https://s3.amazonaws.com'
const backet = BACKET
const serviceFolder = 'service-icons'
const jobsFolder = 'job-pics'

export const getServiceImage = fileName => {
  const iconName = fileName.replace(/\s/g, '%20')

  return `${s3}/${backet}/${serviceFolder}/${iconName}`
}

export const getJobImage = (id, fileName) =>
  `${s3}/${backet}/${jobsFolder}/${id}/${fileName}`

export const getProfileImage = ({ type, src, userId }) => {
  let source = src

  if (_.isString(src)) {
    if (/(data:image\/.*?;base64),/g.exec(src) || src.startsWith('http')) {
      source = src
    } else {
      source = ProfileImagesManager.getSignedURLForProfileImage(
        type,
        src,
        userId,
      )
    }
  } else if (_.isObject(src)) {
    source = null
  }

  return source
}

export const getProfileAvatar = (avatarURL, userId) => {
  if (isUrl(avatarURL)) return avatarURL

  if (_.isString(avatarURL)) {
    return getProfileImage({
      type: profileImageTypes.avatar,
      src: avatarURL,
      userId,
    })
  }

  return null
}
