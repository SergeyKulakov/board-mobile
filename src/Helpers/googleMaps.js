import _ from 'lodash'
import memoize from 'memoize-one'

const googleMaps = 'https://maps.googleapis.com/maps/api/staticmap'
const googleGeocoding = 'https://maps.googleapis.com/maps/api/geocode'
const params = {
  maptype: 'roadmap',
  size: '600x300',
  zoom: '15',
  markers: [
    'color:blue|label:S|40.702147,-74.015794', // example
  ],
  key: 'AIzaSyAOrI9daMiZzf1t2oPscaFD4FpPod4yTR8',
  center: 'BrooklynBridge,New York,NY', // example
}

export const getMapImage = (geolocation = {}) => {
  const { lat, lon } = geolocation
  if (lat && lon) {
    return `${googleMaps}?center=${lat},${lon}&zoom=${params.zoom}&size=${
      params.size
    }&maptype=${
      params.maptype
    }&markers=color:red%7Clabel:S%7C${lat},${lon}&key=${params.key}`
  }

  return `https://maps.googleapis.com/maps/api/staticmap?center=BrooklynBridge,New%20York,NY&zoom=0&size=600x300&maptype=roadmap&key=${
    params.key
  }`
}

export const getLocatinByGeocoding = (lat, lon) => {
  return `${googleGeocoding}/json?latlng=${lat},${lon}&key=${params.key}`
}

export const getGeocodeObg = memoize(geocode => {
  if (_.isString(geocode) && !_.isEmpty(geocode)) {
    const geodata = geocode.split('/')
    return {
      lat: Number(geodata[0]),
      lon: Number(geodata[1]),
    }
  }

  return {}
})
