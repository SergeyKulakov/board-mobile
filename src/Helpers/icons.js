import colors from 'Themes/Colors'
import _ from 'lodash'

export const getNavigationIconStyle = (icon, allSize = 20) => {
  const defaultProps = {
    size: allSize,
    color: colors.blue,
  }

  if (_.isString(icon) || !icon) {
    return {
      name: icon,
      ...defaultProps,
    }
  }

  return {
    type: icon.type,
    name: icon.name,
    color: icon.color || defaultProps.color,
    size: icon.size || defaultProps.size,
  }
}
