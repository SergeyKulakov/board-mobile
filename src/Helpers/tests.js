export const getTestID = value => `[testID="${value}"]`

export const serverErrorObg = {
  error: {
    payload: {
      message: 'message',
      code: 'code',
    },
  },
}
