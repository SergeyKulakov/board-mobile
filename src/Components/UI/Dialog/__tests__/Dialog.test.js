import React from 'react'
import { shallow } from 'enzyme'
import { getTestID } from 'Helpers/tests'

import 'jest-styled-components'

import Component from '../index'

describe('Dialog', () => {
  const defaultProps = {
    editText: 'editText',
    deleteText: 'deleteText',
    onEdit: () => null,
    onDelete: () => null,
    isVisible: true,
    onClose: () => null,
    t: text => text,
  }
  const getDialogRef = () => ({
    current: {
      handleDismiss: jest.fn(),
    },
  })

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('call edit action', () => {
    const props = {
      ...defaultProps,
      onEdit: jest.fn(),
    }

    const wrapper = shallow(<Component {...props} />)

    expect(props.onEdit).not.toHaveBeenCalled()

    wrapper
      .find(getTestID('editButton'))
      .props()
      .onPress()

    expect(props.onEdit).toHaveBeenCalledTimes(1)
  })

  it('call delete action', done => {
    const props = {
      ...defaultProps,
      onDelete: jest.fn(),
    }
    const dialogRef = getDialogRef()

    const wrapper = shallow(<Component {...props} />)

    wrapper.instance().dialog = dialogRef

    wrapper
      .find(getTestID('deleteButton'))
      .props()
      .onPress()

    expect(dialogRef.current.handleDismiss).toHaveBeenCalledTimes(1)
    expect(props.onDelete).not.toHaveBeenCalled()

    setTimeout(() => {
      expect(props.onDelete).toHaveBeenCalledTimes(1)
      done()
    }, 400)
  })

  describe('call back handler', () => {
    it('with isVisible', () => {
      const props = {
        ...defaultProps,
        onClose: jest.fn(),
        isVisible: true,
      }
      const dialogRef = getDialogRef()

      const wrapper = shallow(<Component {...props} />)

      wrapper.instance().dialog = dialogRef

      const response = wrapper.instance().handleBackPress()

      expect(response).toBe(false)

      expect(props.onClose).toHaveBeenCalledTimes(1)

      expect(dialogRef.current.handleDismiss).toHaveBeenCalledTimes(1)
    })

    it('without isVisible', () => {
      const props = {
        ...defaultProps,
        onClose: jest.fn(),
        isVisible: false,
      }
      const dialogRef = getDialogRef()

      const wrapper = shallow(<Component {...props} />)

      wrapper.instance().dialog = dialogRef

      const response = wrapper.instance().handleBackPress()

      expect(response).toBe(true)

      expect(props.onClose).not.toHaveBeenCalled()

      expect(dialogRef.current.handleDismiss).not.toHaveBeenCalled()
    })
  })
})
