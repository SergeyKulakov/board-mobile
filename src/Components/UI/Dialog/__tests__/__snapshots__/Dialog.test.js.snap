// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog render snapshot 1`] = `
<PopupDialog
  dialogAnimation={
    ScaleAnimation {
      "animate": 0,
      "useNativeDriver": true,
    }
  }
  onTouchOutside={[Function]}
  rounded={true}
  visible={true}
  width={0.9}
>
  <DialogActionList
    bordered={true}
    style={null}
  >
    <DialogButton
      activeOpacity={0.6}
      align="center"
      bordered={false}
      disabled={false}
      onPress={[Function]}
      style={null}
      testID="editButton"
      text="editText"
      textStyle={null}
    />
    <DialogButton
      activeOpacity={0.6}
      align="center"
      bordered={false}
      disabled={false}
      onPress={[Function]}
      style={null}
      testID="deleteButton"
      text="deleteText"
      textStyle={null}
    />
  </DialogActionList>
</PopupDialog>
`;
